app/src/main/java/ch/epfl/sdp/lobby/global/GlobalLobbyFragment.kt

```diff
@@ -0,0 +1,45 @@
+package ch.epfl.sdp.lobby.global
+
+import android.content.Context
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ch.epfl.sdp.databinding.FragmentGlobalLobbyBinding
+
+class GlobalLobbyFragment : Fragment() {
+
+    private lateinit var binding: FragmentGlobalLobbyBinding
+    private lateinit var viewAdapter: RecyclerView.Adapter<*>
+    private lateinit var viewManager: RecyclerView.LayoutManager
+    var repo: IGlobalLobbyRepository? = null
```

Comment:
Why is the repository nullable ?
The fragment should get it from the application context that is always available. I think a private val should be used here

Reply:
Because I want to be sure the repo was set, I don't think it's bad in that case

Reply:
Getting it from the activity makes for too much coupling I belive, I'd rather have a fragment that's as independant as possible